src/service/resources-impl.js

```diff
@@ -1327,6 +1327,13 @@ export class Resources {
     // Breadth-first search.
     if (element.classList.contains('-amp-element')) {
       callback(this.getResourceForElement(element));
+      // Also schedule amp-element that is a placeholder for the element.
+      const placeholder = element.getPlaceholder();
+      const hasAmpPlaceholder = (!element.classList.contains('-amp-layout') &&
```

Comment:
Can we move the `-amp-layout` check earlier, to avoid doing a DOM lookup for a placeholder that we may not need.